examples/externalContactsDemo/script.js

```diff
@@ -0,0 +1,269 @@
+                    isAllowed = true;
+                }
+            }
+        });
+
+        return isAllowed;
+    }
+
+    function isPermission(item, targetItem) {
+        let isItem = item === '*' || targetItem === '*';
+        if (!isItem) {
+            isItem = item === targetItem;
+        }
+        return isItem;
+    }
+
+    /**
+     * Determine the embedding PureCloud environment seeded on the query string or
+     * being returned through the OAuth2 Implicit grant state hash param.
+     *
+     * @returns A string indicating the embedding PC env (e.g. mypurecloud.com, mypurecloud.jp); otherwise, null.
+     */
+    function getEmbeddingPCEnv() {
+        let result = null;
+
+        if (window.location.hash && window.location.hash.indexOf('access_token') >= 0) {
+            let oauthParams = extractParams(window.location.hash.substring(1));
+            if (oauthParams && oauthParams.access_token && oauthParams.state) {
+                // OAuth2 spec dictates this encoding
+                // See: https://tools.ietf.org/html/rfc6749#appendix-B
+                let stateSearch = unescape(oauthParams.state);
+                result = extractParams(stateSearch).pcEnvironment;
+            }
+        }
+
+        if (!result && window.location.search) {
+            result = extractParams(window.location.search.substring(1)).pcEnvironment || null;
+        }
+
+        return result;
+    }
+
+    function extractParams(paramStr) {
+        let result = {};
+
+        if (paramStr) {
+            let params = paramStr.split('&');
+            params.forEach(function (currParam) {
+                if (currParam) {
+                    let paramTokens = currParam.split('=');
+                    let paramName = paramTokens[0];
+                    let paramValue = paramTokens[1];
+                    if (paramName) {
+                        paramName = decodeURIComponent(paramName);
+                        paramValue = paramValue ? decodeURIComponent(paramValue) : null;
+
+                        if (!result.hasOwnProperty(paramName)) {
+                            result[paramName] = paramValue;
+                        } else if (Array.isArray(result[paramName])) {
+                            result[paramName].push(paramValue);
+                        } else {
+                            result[paramName] = [result[paramName], paramValue];
+                        }
+                    }
+                }
+            });
+        }
+
+        return result;
+    }
+
+    // Request first three external contacts from the api
+    function requestExternalContacts() {
+        let apiInstance = new platformClient.ExternalContactsApi();
+        let opts = {
+            pageSize: 6,
+            pageNumber: 1
+        };
+        return apiInstance.getExternalcontactsContacts(opts)
+            .then(data => {
+                return data;
+            }).catch(err => {
+                setErrorState(err);
```

Comment:
Catching here actually resolves the promise with undefined which will then throw a reference error in the then handler above.  I would recommend you dont catch here and deal with it above.